README.md

```diff
@@ -70,6 +72,40 @@ cd scripts/user_scripts
 ./train_model.sh
 ```
 
+More details for each of these steps are described more detail in the following sections.
+
+#### Data Preprocessing
+
+This repository uses [Dask](https://docs.dask.org/en/stable/) to load and process data. Our code is configured to load datasets into Dask from Parquet files. If using a different file format, this may need to be changed.
+
+##### Downloading Data
+
+This repository is designed to utilize datasets available through the HuggingFace Hub. There are two ways to download data:
+
+1. [download_data.py](./src/download_data.py), run via the [download_data.sh](./scripts/download_data.sh) script, can download data through the HuggingFace Filesystem. To do this, ensure that the correct parameters are set in the configuration YAML, specifically `hf_filesystem_path`. This must correspond to the correct HF Filesystem path. This method is good for smaller datasets that can easily fit in memory.
+
+2. By cloning the HuggingFace Dataset repo directly, and downloading the necessary data. [download_c4.sh](./scripts/download_c4.sh) exists as an example of this for the C4 dataset. This method is good for very large datasets.
+
+It should be noted that datasets can come in a variety of different formats. Currently, this repo works best with Parquet files. If the data is downloaded in a different format, the code may need to be changed to accomodate.
+
+##### Splitting the Data
+
+After downloading the data, you can split the data into separate train/validation/test splits via the [split_data.py](./src/split_data.py) script, run through [split_data.sh](./scripts/split_data.sh).
+
+Optionally, within [split_data.py](./src/split_data.py), you can specify to shuffle the data while splitting. This is more expensive for larger datasets. Any other preprocessing and pretokenization steps should occur within this step.
```

Comment:
Replace 

"Any other preprocessing and pretokenization steps should occur within this step." 

with

"This and any other preprocessing or pretokenization steps should occur here prior to splitting the data."